packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -892,15 +933,117 @@ impl Store for PostgresDatabase {
         Ok(persisted)
     }
 
-    async fn create_entity() -> Result<(), InsertionError> {
-        todo!()
+    async fn create_entity(
+        &self,
+        entity: &Entity,
+        entity_type_uri: VersionedUri,
+        created_by: AccountId,
+    ) -> Result<EntityId, InsertionError> {
+        let mut transaction = self
+            .pool
+            .begin()
+            .await
+            .report()
+            .change_context(InsertionError)?;
+
+        let entity_id = EntityId::new(Uuid::new_v4());
+        Self::insert_entity(
+            &mut transaction,
+            entity_id,
+            0,
+            entity,
+            entity_type_uri,
+            created_by,
+        )
+        .await?;
+
+        transaction
+            .commit()
+            .await
+            .report()
+            .change_context(InsertionError)?;
+
+        Ok(entity_id)
     }
 
-    async fn get_entity() -> Result<(), QueryError> {
-        todo!()
+    async fn get_entity(&self, entity_id: EntityId) -> Result<Entity, QueryError> {
+        let row = self
+            .pool
+            .fetch_one(
+                sqlx::query(
+                    r#"
+                    SELECT properties
+                    FROM entities
+                    WHERE entity_id = $1 AND version = (
+                        SELECT MAX("version")
+                        FROM entities
+                        WHERE entity_id = $1
+                    );
+                    "#,
+                )
+                .bind(entity_id),
+            )
+            .await
+            .report()
+            .change_context(QueryError)
+            .attach_printable(entity_id)?;
+
+        Ok(serde_json::from_value(row.get(0))
+            .report()
+            .change_context(QueryError)?)
     }
 
-    async fn update_entity() -> Result<(), UpdateError> {
-        todo!()
+    async fn update_entity(
+        &self,
+        entity_id: EntityId,
+        entity: &Entity,
+        entity_type_uri: VersionedUri,
+        updated_by: AccountId,
+    ) -> Result<(), UpdateError> {
+        let mut transaction = self
+            .pool
+            .begin()
+            .await
+            .report()
+            .change_context(UpdateError)?;
+
+        // TODO: OPTIM: Combine version look-up and insertion into one query
+        let entity_version: i32 = self
```

Comment:
I'm not sure about using an i32 for an entity version. @thehabbos007 @benwerner01 do you have thoughts? I think an XID or timestamp might be good, should save us this look-up too.

Reply:
We discussed this on-call, timestamp seems like a good option instead as we don't need the ID to be as important as versions for the ontology types, and it means one less query for fetching the latest version in the DB (we can also just have the DB add the timestamp)